### ISSUE TITLE
Add Multi-Horizon Time Series Classification Support

### ISSUE THREAD

**Issue by eetulauri** [unit8co/darts#2315]
Darts has established itself as a premier time series forecasting library. Adding multi-horizon time series classification support would solidify its position and significantly benefit researchers and practitioners alike. This feature aligns with previous discussions and interest, as seen in issue #1473.

Use Case Example:

Traffic congestion prediction: Classifying whether traffic congestion is likely to occur over multiple upcoming time intervals (e.g., next 15 minutes, next hour) based on historical traffic patterns.

Desired Functionality

- Input: Time series data and desired classification horizon.
- Output: Class labels for each time series segment across the specified horizon.

Potential Approaches

- Incorporate classifiers like LightGBMClassifier, enabling them to produce class label predictions for multiple future time steps within the Darts framework.

**Comment by madtoinou** [comment]
Hi @eetulauri,

This is on Darts' roadmap but we would like to add conformal prediction before tackling the classification features as it would require some work for the deep learning models.

As described in the linked issue, you can easily use "Classifier" variant of supported model by creating a new class and overwritting the "model" attribute.

You can also use classifier models from sklearn thanks to the `RegressionModel` class;

```python
import numpy as np
from darts import TimeSeries
from darts.models import RegressionModel
from sklearn.ensemble import RandomForestClassifier

ts = TimeSeries.from_values(np.random.randint(0,10,100))
model = RegressionModel(
    lags=4,
    model=RandomForestClassifier()
)
model.fit(ts)
model.predict(10)
```

**Comment by tuomijal** [comment]
Hi @madtoinou,

Has there been any progress on this? Let me know if I can take a look at this and be of assistance.

**Comment by madtoinou** [comment]
Hi @tuomijal,

Conformal prediction is scheduled to be part of the "holidays release" and we then plan to clearly define the priorities of the roadmap. If you have some bandwidth to work on this feature, it's more than welcome, having some code for it will definitely improve the quality of the conversation. WDYT @dennisbader ?

**Comment by tuomijal** [comment]
Thank you for the quick response and your continued support on the project, @madtoinou and @dennisbader! 

I might have some time to dive into it after the holidays. Is there anything I should know before I give it a shot?

**Comment by tuomijal** [comment]
A classifier outputs $d\geq2$ dimensions for each horizon $h\geq1$ so I guess this would require touching TimeSeries object as well?

**Comment by madtoinou** [comment]
Why would the classifier assign several different classes to a single time-step? You could eventually leverage the sample dimension.

**Comment by tuomijal** [comment]
I mean that the output dimensions of a classifier is equal to the number of classes.

**Comment by wouterbles** [comment]
@tuomijal also interested in this issue, have you made any progress so far?

**Comment by tuomijal** [comment]
Not yet unfortunately
